evalml/tests/pipeline_tests/test_pipelines.py

```diff
@@ -244,7 +245,7 @@ def test_make_pipeline_problem_type_mismatch():
         make_pipeline(pd.DataFrame(), pd.Series(), Transformer, ProblemTypes.MULTICLASS)
 
 
-def test_make_pipeline_from_components():
+def test_make_pipeline_from_components(X_y_binary):
```

Comment:
@christopherbunn this test is great! What are some other cases you could cover? Here's a few thoughts on edge cases:
def make_pipeline_from_components(component_instances, problem_type, custom_name=None):
* What if `component_instances` contains class references instead of instances? Or generally things which aren't evalml component instances?
* What if `problem_type` is invalid?
* What if `custom_name` is not a string?

In all of those cases, I'd hope we'd get some sort of exception when we try to define the pipeline template class. As it stands, because our `PipelineBaseMeta` metaclass is pretty simple and doesn't check these fields, I bet no error comes up for some, so perhaps `make_pipeline_from_components` should error out instead.